src/core/execute.c

```diff
@@ -2295,6 +2294,12 @@ static int compile_bind_mounts(
 
         assert(h == n);
 
+        if (!strv_isempty(empty_directories) && !strv_isempty(context->empty_directories)) {
```

Comment:
I admit I haven't spent much time looking at this code, but at a casual glance shouldn't context->empty_directories still be applied when empty_directories is empty at this point?  It looks like a non-empty context->empty_directories will just be ignored by this function if empty_directories is empty.  Is that intentional?

Reply:
yeah, i think @vcaputo is right